main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripAndShapeDistanceValidator.java

```diff
@@ -32,55 +35,75 @@
  */
 @GtfsValidator
 public class TripAndShapeDistanceValidator extends FileValidator {
-
   private final GtfsTripTableContainer tripTable;
-
   private final GtfsStopTimeTableContainer stopTimeTable;
-
+  private final GtfsStopTableContainer stopTable;
   private final GtfsShapeTableContainer shapeTable;
+  private final double DISTANCE_THRESHOLD = 11.1; // distance in meters
 
   @Inject
   TripAndShapeDistanceValidator(
       GtfsTripTableContainer tripTable,
       GtfsStopTimeTableContainer stopTimeTable,
+      GtfsStopTableContainer stopTable,
       GtfsShapeTableContainer shapeTable) {
     this.tripTable = tripTable;
     this.stopTimeTable = stopTimeTable;
     this.shapeTable = shapeTable;
+    this.stopTable = stopTable;
   }
 
   @Override
   public void validate(NoticeContainer noticeContainer) {
-    shapeTable
-        .byShapeIdMap()
+    tripTable
+        .getEntities()
         .forEach(
-            (shapeId, shape) -> {
-              double maxShapeDist =
+            trip -> {
+              String shapeId = trip.shapeId();
+              // Get distance for trip
+              int nbStopTimes = stopTimeTable.byTripId(trip.tripId()).size();
+              if (nbStopTimes == 0) {
+                return;
+              }
+              GtfsStopTime lastStopTime =
+                  stopTimeTable.byTripId(trip.tripId()).get(nbStopTimes - 1);
+              GtfsStop stop = stopTable.byStopId(lastStopTime.stopId()).orElse(null);
+              if (stop == null) {
+                return;
+              }
+              double maxStopTimeDist = lastStopTime.shapeDistTraveled();
+
+              // Get max shape distance for trip
+              GtfsShape maxShape =
```

Comment:
I thought shape was also guaranteed to have the greatest distance on the last row. If not there's a error notice for that. Do we need to look for the greatest?

Reply:
It is indeed the case. However, based on my analysis, this aspect does not affect performance the same way iterating through `stop_times` does. Therefore, I chose to be a bit more thorough in this step and include this check.